src/components/tx/DecodedTx/index.tsx

```diff
@@ -75,7 +75,7 @@ const DecodedTx = ({
 
   if (!decodedData) return null
 
-  const amount = tx?.data.value ? formatVisualAmount(tx.data.value, chain?.nativeCurrency.decimals) : '0'
+  const amount = tx?.data.value ? safeFormatUnits(tx.data.value, chain?.nativeCurrency.decimals) : '0'
```

Comment:
I've tested this and can confirm that for large numbers it fixes the `NaN` issue but it seems to also change the formatting for smaller numbers e.g. sending `468651  ETH` via transaction builder results in `468,651` if `formatVisualAmount` is used and `468.651` if `safeFormatUnits` is used. My `navigator.language` is set to `de`.

Reply:
I will refactor it and comment that `SendAmountBlock` always expects the WEI amount. And then move all formatting inside there.